src/depends/libDatabase/LevelDB.cpp

```diff
@@ -15,610 +15,493 @@
  * along with this program.  If not, see <https://www.gnu.org/licenses/>.
  */
 
-#include <string>
-
-
 #include "LevelDB.h"
 #include "common/Constants.h"
-#include "depends/common/Common.h"
 #include "depends/common/CommonData.h"
-#include "depends/common/FixedHash.h"
 #include "libUtils/Logger.h"
 
 using namespace std;
 
-void LevelDB::log_error(leveldb::Status status) const
-{
-    if(!status.IsNotFound())
-    {
-     LOG_GENERAL(WARNING, "LevelDB " << m_dbName << " status is not OK - " << status.ToString());
-    }
-}
+namespace {
 
-LevelDB::LevelDB(const string& dbName, const string& path, const string& subdirectory)
-{
-    this->m_subdirectory = subdirectory;
-    this->m_dbName = dbName;
-    this->m_db = NULL;
+template <typename ArgT>
+std::string LookupImpl(const std::shared_ptr<leveldb::DB> db,
+                       const std::string& dbName, ArgT&& arg,
+                       bool* found = nullptr) {
+  std::string value;
+  if (!db) {
+    LOG_GENERAL(WARNING, "LevelDB " << dbName << " isn't available");
+    if (found) {
+      *found = false;
+    }
+    return value;
+  }
 
-    if(!(std::filesystem::exists(path)))
-    {
-        LOG_GENERAL(WARNING, path + " does not exist");
-        return;
+  auto s = db->Get(leveldb::ReadOptions(), std::forward<ArgT>(arg), &value);
+  if (!s.ok()) {
+    if (!s.IsNotFound()) {
+      LOG_GENERAL(WARNING, "LevelDB " << dbName << " status is not OK - "
+                                      << s.ToString());
+    }
```

Comment:
I'm not sure this makes sense. Would `s.IsNotFound()` be `true` if the DB exists, but the key (`arg`) doesn't exist in it?

Reply:
It would; this is backward compatible with the current implementation. Haven't changed anything there.